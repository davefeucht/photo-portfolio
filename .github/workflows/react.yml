# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions

name: React CI/CD

on:
  push:
    branches: [ master ]
  pull_request:
    branches: [ master ]

jobs:
  build:

    runs-on: ubuntu-latest

    strategy:
      matrix:
        node-version: [18.x]

    steps:
    - name: Checkout code
      uses: actions/checkout@v2

    - name: Install Node.js ${{ matrix.node-version }}
      uses: actions/setup-node@v1
      with:
        node-version: ${{ matrix.node-version }}

    - name: Enable corepack
      run: corepack enable

    - name: Install dependencies
      run: yarn install

    - name: Build project
      run: npm run build --if-present

    - name: Run tests
      run: npm test
