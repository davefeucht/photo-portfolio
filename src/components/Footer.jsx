/*****************
* Footer component implements the footer of the application
*****************/

import React from "react";
import StyledFooter from "./styledComponents/StyledFooter.jsx";

export default function Footer (props) {

    return (
      <StyledFooter>
        Theme Copyright @2018 Dave Feucht  
      </StyledFooter>
    );
}