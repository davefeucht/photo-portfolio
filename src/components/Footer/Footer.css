.footer {
  color: #FFFFFF;
  grid-area: footer;
  background: rgba(0,0,0,.75); 
  display: grid;
  grid-template-colums: 100%;
  grid-template-rows: 100%;
  align-items: center;
  padding: 5px;
}