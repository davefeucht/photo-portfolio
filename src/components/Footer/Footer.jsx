/** ***************
* Footer component implements the footer of the application
**************** */

import './Footer.css';

import React from 'react';

function Footer() {
    return (
        <div className="footer">
            Theme Copyright @2021 Dave Feucht
        </div>
    );
}

Footer.displayName = 'Footer';

export default Footer;
