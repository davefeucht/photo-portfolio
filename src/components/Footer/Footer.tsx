/** ***************
* Footer component implements the footer of the application
**************** */

import './Footer.css';

import * as React from 'react';

const Footer = () => {
    return (
        <div className="footer">
            Theme Copyright @2023 Dave Feucht
        </div>
    );
};

Footer.displayName = 'Footer';

export default Footer;
