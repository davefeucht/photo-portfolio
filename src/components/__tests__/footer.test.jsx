import { render } from '@testing-library/react';
import React from 'react';
import { act } from 'react-dom/test-utils';
import { MemoryRouter } from 'react-router-dom';

import Footer from '../Footer/Footer';

test('Footer displays', async () => {
    let container;
    await act(async () => {
        container = render(
            <MemoryRouter>
                <Footer />
            </MemoryRouter>
        ).container;
    });
    expect(container.firstChild).toMatchSnapshot();
});
