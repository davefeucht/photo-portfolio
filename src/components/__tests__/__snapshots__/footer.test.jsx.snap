// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer displays 1`] = `
<div
  class="footer"
>
  Theme Copyright @2023 Dave Feucht
</div>
`;
